src/Raven.Server/Documents/Indexes/Persistence/Corax/JintCoraxDocumentConverter.cs

```diff
@@ -312,31 +312,39 @@ private IndexField GetFieldObjectForProcessing(in string propertyAsString, Curre
 
         var valueAsObject = iterator.Current.AsObject();
 
-        return TryDetectDynamicFieldCreation(propertyAsString, valueAsObject, field, indexingScope) is not null
+        return TryDetectDynamicFieldCreation(propertyAsString, valueAsObject, ref field, indexingScope) is not null
                || valueAsObject.HasOwnProperty(JavaScriptFieldName.SpatialPropertyName);
     }
 
 
-    private static JsValue TryDetectDynamicFieldCreation(string property, ObjectInstance valueAsObject, IndexField field, CurrentIndexingScope scope)
+    private JsValue TryDetectDynamicFieldCreation(string property, ObjectInstance valueAsObject, ref IndexField field, CurrentIndexingScope scope)
     {
         //We have a field creation here _ = {"$value":val, "$name","$options":{...}}
         if (!valueAsObject.HasOwnProperty(JavaScriptFieldName.ValuePropertyName))
             return null;
 
         var value = valueAsObject.GetOwnProperty(JavaScriptFieldName.ValuePropertyName).Value;
         PropertyDescriptor nameProperty = valueAsObject.GetOwnProperty(JavaScriptFieldName.NamePropertyName);
+
+        string fieldName;
+        FieldIndexing? fieldIndexing = null;
+        FieldStorage? fieldStorage = null;
+        
         if (nameProperty != null)
         {
             var fieldNameObj = nameProperty.Value;
             if (fieldNameObj.IsString() == false)
                 throw new ArgumentException($"Dynamic field {property} is expected to have a string {JavaScriptFieldName.NamePropertyName} property but got {fieldNameObj}");
 
-            field.Name = fieldNameObj.AsString();
+            fieldName = fieldNameObj.AsString();
             field.Id = CoraxLib.Constants.IndexWriter.DynamicField;
         }
         else
         {
-            field.Name = property;
+#if DEBUG
+            throw new InvalidOperationException($"{nameof(nameProperty)} should not be a null.");
+#endif
+            fieldName = property;
```

Comment:
Please check `Unreachable code detected` warning here

Reply:
I guess something had to change in Jint itself because this particular code is old and now it seems it cannot be null.

Reply:
We have upgraded the Jint meanwhile. You need to verify that.

Reply:
The `GetOwnProperty` in Jint has following description:
https://github.com/sebastienros/jint/blob/65a5ba3afb8d50e128e635a148db0ebe28deba46/Jint/Native/Object/ObjectInstance.cs#L414-L420

I guess instead of comparing `!= null` we have to check it like this: `nameProperty.Value.IsUndefined() == false`